lib/block-supports/layout.php

```diff
@@ -116,28 +122,30 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 			 * by custom css.
 			 */
 			if ( ! empty( $layout['justifyContent'] ) && array_key_exists( $layout['justifyContent'], $justify_content_options ) ) {
-				$style .= "$selector {";
-				$style .= "justify-content: {$justify_content_options[ $layout['justifyContent'] ]};";
-				$style .= '}';
+				$declarations[ $selector ]['justify-content'] = $justify_content_options[ $layout['justifyContent'] ];
 			}
 
 			if ( ! empty( $layout['verticalAlignment'] ) && array_key_exists( $layout['verticalAlignment'], $vertical_alignment_options ) ) {
-				$style .= "$selector {";
-				$style .= "align-items: {$vertical_alignment_options[ $layout['verticalAlignment'] ]};";
-				$style .= '}';
+				$declarations[ $selector ]['align-items'] = $vertical_alignment_options[ $layout['verticalAlignment'] ];
 			}
 		} else {
-			$style .= "$selector {";
-			$style .= 'flex-direction: column;';
+			$declarations[ $selector ]['flex-direction'] = 'column';
+			$declarations[ $selector ]['align-items']    = 'flex-start';
 			if ( ! empty( $layout['justifyContent'] ) && array_key_exists( $layout['justifyContent'], $justify_content_options ) ) {
-				$style .= "align-items: {$justify_content_options[ $layout['justifyContent'] ]};";
-			} else {
-				$style .= 'align-items: flex-start;';
+				$declarations[ $selector ]['align-items'] = $justify_content_options[ $layout['justifyContent'] ];
 			}
-			$style .= '}';
 		}
 	}
 
+	$style = '';
+	foreach ( $declarations as $css_selector => $item_declarations ) {
+		if ( empty( $item_declarations ) ) {
+			continue;
+		}
+		$declarations_obj = new WP_Style_Engine_CSS_Declarations_Gutenberg( $item_declarations );
```

Comment:
Would this be a good place to use our (future) store?

Beware! Pseudo code:

```php
// In layout.php
foreach ( $declarations as $css_selector => $item_declarations ) {
     wp_style_engine_register_block_supports_css( $css_selector, $item_declarations );
}

// Somewhere in the style engine
function wp_style_engine_register_block_supports_css( $css_selector, $item_declarations ) {
      $declarations_obj = new WP_Style_Engine_CSS_Declarations_Gutenberg( $item_declarations );
      $some_previously_instantiated_store->register( $css_selector, $declarations_obj );
}

// Somewhere else in the style engine
$stylesheet =  some_compile_and_bundle_css_rules_from_store_function();
add_action(
      'some_action,
      function () {
            wp_register_style( 'block-supports', false, array(), true, true );
             wp_add_inline_style( 'block-supports', $stylesheet );
             wp_enqueue_style( 'block-supports' );
      },
      $priority
);

```

Reply:
Another thought I had, and sorry for spamming this PR, was to have a Renderer_Class that extends a store class.

The renderer class would have access to its own store and be able to fetch, process and compile stored styles, then render them to the page.

E.g., 

```php
class WP_Style_Engine_Renderer extend WP_Style_Engine_Store {}

// Somewhere in the style engine.
$block_supports_renderer = new WP_Style_Engine_Renderer();
$block_supports_renderer->enqueue();

// Add a bunch of rules
$block_supports_renderer->add( $selector, $rules );
```

Reply:
> Would this be a good place to use our (future) store?

Definitely! If there is a selector then using the store would be preferable as it can include many things like optimizations, caching and whatnot. The Declarations object will be more lower-level and in the final implementation of the Style Engine I expect it to be used sparingly, mostly when there is no selector (so for inline styles like `<div style="...">`), or for cases where we want to bypass the store for some reason.

> Another thought I had, and sorry for spamming this PR, was to have a Renderer_Class that extends a store class.
> 
> The renderer class would have access to its own store and be able to fetch, process and compile stored styles, then render them to the page.

I agree, it would make sense to have a Renderer object... Though I'm not sure it should _extend_ the Store. 🤔
It depends on what we want it to do I suppose. If the Renderer's job is to get a store, calculate/optimize styles and print them, then we'll need to be able to pass a specific store to it (since there will be multiple stores).
So something like this would also make sense:
```php
$store    = WP_Style_Engine_Store::get_store( 'whatever' );
$renderer = new WP_Style_Engine_Renderer( $store );
$renderer->enqueue();
```

Reply:
>  If the Renderer's job is to get a store, calculate/optimize styles and print them, then we'll need to be able to pass a specific store to it (since there will be multiple stores).
So something like this would also make sense

Passing the store on instantiation would also work, and is probably the better way to go.

You're right, it depends on how we want to do things. 

I was making a huge leap. I just wanted to write it down in case I forgot 🤣